### PR TITLE
Add anonymous support to SDK

### PR DESCRIPTION
**Pull request by BernardGatt**
Issue: https://linear.app/customerio/issue/INAPP-12353/gist-sdk-poll-for-display-in-app-broadcast

Note, the change is currently deployed on beta: https://code.gist.build/web/beta/gist.min.js

```js
Gist.setup({
    siteId: "XXX",
    dataCenter: "us", 
    env: "dev", 
    logging: true, 
    useAnonymousSession: true
});

Gist.setUserLocale("en-GB");
```

### CODE REVIEW

File: src/managers/message-broadcast-manager.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { getHashedUserToken } from './user-manager';
+
+const broadcastsLocalStoreName = "gist.web.message.broadcasts";
+
+export async function updateBroadcastsLocalStore(messages) {
+  const messageBroadcastLocalStoreName = await getMessageBroadcastLocalStoreName();
+  if (!messageBroadcastLocalStoreName) return;
+
+  const expiryDate = new Date();
+  expiryDate.setDate(expiryDate.getDate() + 30);
```
Comment (karngyan) [inline src/managers/message-broadcast-manager.js:12]: May be we move the 30 out to a constant.

File: src/managers/message-user-queue-manager.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+import { setKeyWithExpiryToLocalStore, getKeyFromLocalStore, setKeyToLocalStore } from '../utilities/local-storage';
+import { getHashedUserToken } from './user-manager';
+
+const messageQueueLocalStoreName = "gist.web.message.user";
+const MESSAGES_LOCAL_STORE_CACHE_IN_MINUTES = 60000 * 60;
```
Comment (karngyan) [inline src/managers/message-user-queue-manager.js:5]: Just curious why we have different naming styles on constants (camelCase vs UPPER_CASE ?)

File: src/utilities/local-storage.js [hunk 3]
```diff
@@ -1,3 +1,13 @@
+export function setKeyToLocalStore(key, value) {
+    let maxExpiryDate = new Date();
+    maxExpiryDate.setDate(maxExpiryDate.getDate() + 365);
```
Comment (karngyan) [inline src/utilities/local-storage.js:3]: another constant

**Review by karngyan — APPROVED** [review]

File: src/managers/message-user-queue-manager.js (hunk 2 above)
Comment (BernardGatt) [reply src/managers/message-user-queue-manager.js]: Addressed this in the last commit.

**Pull request merged**